vllm/config.py

```diff
@@ -2909,31 +2908,30 @@ def init_backend(self, vllm_config: "VllmConfig") -> Union[str, Callable]:
         from vllm.compilation.backends import VllmBackend
         return VllmBackend(vllm_config)
 
-    def init_with_cudagraph_sizes(self, sizes_to_specialize: List[int]):
+    def init_with_cudagraph_sizes(self,
+                                  cudagraph_capture_sizes: List[int]) -> None:
         """To complete the initialization of config,
         we need to know the cudagraph sizes."""
 
-        if self.cudagraph_capture_sizes is None:
-            self.capture_sizes = sizes_to_specialize
+        if self.capture_sizes is None:
+            self.capture_sizes = cudagraph_capture_sizes
         else:
-            self.capture_sizes = self.cudagraph_capture_sizes
             logger.info(("cudagraph sizes specified by model runner"
                          " %s is overridden by config %s"),
-                        sizes_to_specialize, self.cudagraph_capture_sizes)
-
-        if self.candidate_compile_sizes is None:
-            self.candidate_compile_sizes = []
-        self.compile_sizes = [
-            x for x in self.candidate_compile_sizes if x in self.capture_sizes
-        ]
-        ignored_sizes = [
-            x for x in self.candidate_compile_sizes
-            if x not in self.capture_sizes
-        ]
-        if ignored_sizes:
-            logger.warning(("candidate_compile_sizes %s are ignored "
-                            "because they are not cudagraph capture sizes."),
-                           ignored_sizes)
+                        cudagraph_capture_sizes, self.capture_sizes)
+
+        computed_compile_sizes = []
+        if self.compile_sizes is not None:
+            for x in self.compile_sizes:
+                if isinstance(x, str):
+                    assert x == "cudagraph", \
+                    "Unrecognized size type in compile_sizes, " \
+                    f"expect 'cudagraph', got {x}"
+                    computed_compile_sizes.extend(self.capture_sizes)
```

Comment:
Should we enforce for a single occurrence of "cudagraph" in the list?  Otherwise, it looks like it is possible to have multiple instances of the same size?

Reply:
de-duplication is done in [39b9245](https://github.com/vllm-project/vllm/pull/12243/commits/39b9245f77d4e834c6b950affed581cd79ae32f4) .